### ISSUE TITLE
[TASK] Block UI elements like delete while drag&drop action is processed in the page view

### ISSUE THREAD

**Issue by BenjaminBeck** [FluidTYPO3/flux#757]
It just happened to a editor that he unintendedly deleted a content element.
He moved the content element outside of a grid and then deleted the grid.
I think the ajax request for moving the element was not finished when he pressed the grids deleted button.

Because of this the element outside the grid also got deleted and lost relation to its a parent element.

A solution could be to block depending action buttons until the ajax request is successfully finished.

The TYPO3 system this happened with is pretty slow atm - so one could argue that this would happen very unlikely happen with a "normal" system...

**Comment by NamelessCoder** [comment]
@BenjaminBeck this is very unfortunate indeed. We can't mitigate this since there is an explicit comment in exactly the piece of (JS) code which handles the droptarget event: it explains that no feedback ever comes from the server and that it is simply ignored. I am not sure if a better solution would be to block the script while it waits for a response or timeout (the newest CMS 7 dev branch seems to do this - it locks for a good while when dropping elements), but you might want to try suggesting that as a fix for the 6.2 branch of TYPO3. In almost all cases this wouldn't cause a problem but you are completely correct that it would do so, if the server has a hard time processing requests in a timely fashion. It might cause a delete (which happens fast and reloads the page, thus cancelling the XHR!) to happen before a move instruction (which usually takes a while longer and has zero visual feedback).

TL;DR: if there is a fix to be done, it has to be in the core - either the server-side script that replies after content is moved or the JS that handles the dispatching of said request.
